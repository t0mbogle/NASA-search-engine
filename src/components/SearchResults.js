import React from "react";
import "../styles/searchresults.css";
import PropTypes from "prop-types";

const SearchResults = ({ results }) => {
  if(!results.length) {
    return(<p className="no-search">No Results found!</p>);
    // If results/searchResults prop has a length map over it 
    // and only return the URL of those mapped images, from the getImages function
  } else {

    return(
      <>
        <p className="success-search">Search Results:</p>
        <div className="search-results">
          {results.map((image) => (
            <img className="card-name" src={image} alt="spaceImage" />
          ))}
        </div>
      </>
    );
  }
};

export default SearchResults;

SearchResults.propTypes = {
  results: PropTypes.arrayOf(
    PropTypes.string).isRequired,
};
