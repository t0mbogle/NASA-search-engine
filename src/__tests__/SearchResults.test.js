import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
    const validProps = {
        results: [
          'https://images-assets.nasa.gov/image/PIA03606/PIA03606~thumb.jpg',
          'https://images-assets.nasa.gov/image/GSFC_20171208_Archive_e000053/GSFC_20171208_Archive_e000053~thumb.jpg'
        ],
    };

    it("renders correctly", () => {
        const { asFragment } = render(
          <SearchResults
            results={validProps.results}
          />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});