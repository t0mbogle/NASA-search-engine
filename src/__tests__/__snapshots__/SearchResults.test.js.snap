// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <p
    class="success-search"
  >
    Search Results:
  </p>
  <div
    class="search-results"
  >
    <img
      alt="spaceImage"
      class="card-name"
      src="https://images-assets.nasa.gov/image/PIA03606/PIA03606~thumb.jpg"
    />
    <img
      alt="spaceImage"
      class="card-name"
      src="https://images-assets.nasa.gov/image/GSFC_20171208_Archive_e000053/GSFC_20171208_Archive_e000053~thumb.jpg"
    />
  </div>
</DocumentFragment>
`;
